src/dove/models/price_taker/rulelib.py

```diff
@@ -36,39 +36,56 @@
     from dove.core import Storage
 
 
-# Transfer Constraints (Converters)
-def transfer_rule(m: pyo.ConcreteModel, cname: str, t: int) -> pyo.Expression:
+def transfer_block_rule(b: pyo.Block, cname: str, t: int) -> None:
     """
-    Create a PyomoExpression for a price-taker component's transfer rule.
+    Sets up a block that handles resource transfer for components.
 
-    This function implements the conservation rule for price-taker components,
-    applying the component-specific transfer function to relate inputs to outputs
-    at the specified time step.
+    The function creates a Pyomo Block that is specific to a non-storage component. It finds the
```

Comment:
I was trying to step through this in debug mode, correct me if I'm wrong but this isn't necessarily creating a Pyomo Block. From the stack trace, looks like its something called in the process of building the Pyomo Block; it accepts the Pyomo Block and the build process naturally iterates over indices (which just so happen to be components and time)?